tests/unit/test_jim.py

```
@@ -558,3 +563,195 @@ def test_sample_raises_on_3d_initial_position(self, basic_jim, monkeypatch):
+            Jim(
+                likelihood=NaNLikelihood(),
+                prior=prior,
+                rng_key=jax.random.key(0),
+                n_chains=5,
+                n_local_steps=2,
+                n_global_steps=2,
+                global_thinning=1,
+            )
+
+    def test_some_nan_posterior_warns(self, caplog):
+        """1–5 NaN log-posteriors out of 10 issues a warning but no error."""
```

Ruff (0.15.11)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)